sbi/inference/snre/snre_c.py

```diff
@@ -185,26 +185,21 @@ def _loss(
             loggamma + logits_joint[:, 0] - torch.logsumexp(denominator_joint, dim=-1)
         )
 
-        # relative weights. p_marginal := p_0, and p_joint := p_K from the notation.
+        # relative weights. p_marginal := p_0, and p_joint := p_K * K from the notation.
         p_marginal, p_joint = self._get_prior_probs_marginal_and_joint(
             num_classes, gamma
         )
-        return -torch.mean(
-            p_marginal * log_prob_marginal + p_joint * num_classes * log_prob_joint
-        )
+        return -torch.mean(p_marginal * log_prob_marginal + p_joint * log_prob_joint)
 
     @staticmethod
-    def _get_prior_probs_marginal_and_joint(
-        num_classes: int, gamma: float
-    ) -> Tuple[float, float]:
+    def _get_prior_probs_marginal_and_joint(gamma: float) -> Tuple[float, float]:
         """Return a tuple (p_marginal, p_joint) where `p_marginal := `$p_0$,
-        `p_joint := `$p_K$.
+        `p_joint := `$p_K \cdot K$.
 
         We let the joint (dependently drawn) class to be equally likely across K
         options. The marginal class is therefore restricted to get the remaining
         probability.
         """
-        assert num_classes >= 1
-        p_joint = gamma / (1 + gamma * num_classes)
-        p_marginal = 1 / (1 + gamma * num_classes)
+        p_joint = gamma / (1 + gamma)
```

Comment:
I see, the point is that `p_K = 1 / K * gamma / (1 + gamma)` so that in `p_joint = K * p_K`,  `1/K` cancels out?

Reply:
I am changing the definition of p_joint here to be K * p_K. Then we get this cancelation you describe. (As written in the paper).

Reply:
see page 4 of https://arxiv.org/pdf/2210.06170.pdf